### PR TITLE
Feat archive all

### PR DESCRIPTION
**Pull request by ingkebil**
Archive cases based on scout input.

### CODE REVIEW

File: moneypenny/update/cli/archive.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import click
+from scout.ext.backend import MongoAdapter
+
+from ..actions.archive import process as process_archive
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.pass_context
```
Comment (robinandeer) [inline moneypenny/update/cli/archive.py]: nice and clean CLI function!

File: moneypenny/update/actions/archive.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                except NoResultFound:
+                    logger.debug("skipping sample: %s", sample_id)
+                    continue
+                sample_obj.delivered_at = scout_case.updated_at
+
+            #case_obj.is_checked = scout_case.analysis_checked
+            scout_status = scout_case.status
+
+            if scout_status in ('archived', 'solved'):
+                if scout_case.created_at > CHECKED_CUTOFF or \
```
Comment (robinandeer) [inline moneypenny/update/actions/archive.py]: consider breaking this long if statement into multiple lines and store each result in a variable!

File: moneypenny/update/actions/archive.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import click
+import datetime
+from sqlalchemy.orm.exc import NoResultFound
+
+from ...flow import actions
```
Comment (robinandeer) [inline moneypenny/update/actions/archive.py]: this is on the verge I think 😉 
sometimes it's clearer just to do an import from `from moneypenny.flow import actions`

**Comment by robinandeer** [comment]
How did this play in with the archive jobs added in the `jobs` step during upload?

File: moneypenny/update/actions/archive.py (hunk 3 above)
Comment (ingkebil) [inline moneypenny/update/actions/archive.py]: Aha, makes sense and good to know!

**Comment by ingkebil** [comment]
I thought there were no jobs being added to the queue so far, is that wrong?

**Comment by ingkebil** [comment]
Aha, I see now.

**Comment by ingkebil** [comment]
In `moneypenny/update/actions/jobs.py`:

```
    # cases marked to be archived
    archives = db.to_archive()
    for case in archives:
        dispatch_archive(db, case.id)
        log.info("queued %s for archiving", case.id)
```

This code takes into account the cutoff and doesn't queue jobs for archiving before the cutoff date.
The above and my code does not play nice together and multiple jobs would be queued for the same case.

**Comment by ingkebil** [comment]
I guess we remove the code in `jobs.py`?

**Comment by ingkebil** [comment]
Well, we don't _need_ to remove it, I haven't added it the `all` action of `update`, so `archive` you could run separately.

**Comment by ingkebil** [comment]
There is something wrong with this code. Somehow the part where a case is added to the queue for archiving is missing :confused:

**Comment by ingkebil** [comment]
It seems we have 209 cases ready for archiving :D

**Pull request merged**